### PR TITLE
Add BGFX shader compilation

### PR DESCRIPTION
**Pull request by lmariscal**
Detect all `*.sc` shader files inside the `res/shaders` folder. It respects the correct directory structure, so it is possible to have a `varying.def.sc` per shader program.  
It adds by default as include path the `res/shaders`, so all the shaders are able to include other `*.sh` from this same path.

Remove examples and their compilation from BGFX.

Add an example for `cube.sc` for both vertex and fragment shaders, with their respective `varying.def.sc`.

### CODE REVIEW

File: core/CMakeLists.txt [hunk 1]
```diff
@@ -50,3 +52,35 @@ foreach(LIB ${PRIVATE_LIBS})
...
+    elseif("${SHADER_NAME}" MATCHES "fs_*")
+        message(STATUS "Found fragment shader ${SHADER_NAME}")
+        set(SHADER_TYPE FRAGMENT)
+    elseif("${SHADER_NAME}" MATCHES "cs_*")
+        message(STATUS "Found compute shader ${SHADER_NAME}")
+        set(SHADER_TYPE COMPUTE)
+    else()
+        message(STATUS "Found & Skipped unknown shader ${SHADER_NAME}")
+        continue()
+    endif()
```
Comment (BertTDev) [inline core/CMakeLists.txt:77]: Are we expecting to use other shader types later and adding them or expecting to only use these three?

File: res/shaders/cubes/varying.def.sc [hunk 2]
```diff
@@ -0,0 +1,4 @@
+vec4 v_color0    : COLOR0    = vec4(1.0, 0.0, 0.0, 1.0);
```
Comment (BeardyKing) [inline res/shaders/cubes/varying.def.sc:1]: unsure why v_ & a_ should be v_ & f_

**Review by BeardyKing — APPROVED** [review]

File: core/CMakeLists.txt [hunk 3]
```diff
@@ -50,3 +52,35 @@ foreach(LIB ${PRIVATE_LIBS})
...
+    if("${SHADER_NAME}" MATCHES "vs_*")
+        message(STATUS "Found vertex shader ${SHADER_NAME}")
+        set(SHADER_TYPE VERTEX)
+    elseif("${SHADER_NAME}" MATCHES "fs_*")
+        message(STATUS "Found fragment shader ${SHADER_NAME}")
+        set(SHADER_TYPE FRAGMENT)
+    elseif("${SHADER_NAME}" MATCHES "cs_*")
+        message(STATUS "Found compute shader ${SHADER_NAME}")
+        set(SHADER_TYPE COMPUTE)
+    else()
```
Comment (BertTDev) [inline core/CMakeLists.txt:74]: Are we expecting to use other shader types later and adding them or expecting to only use these three?

Comment (lmariscal) [reply core/CMakeLists.txt:74]: At the moment we are only expecting to use these shader types. We might expand in the future, geometry shaders for example, but to start building the renderer pipeline let's start with these.

**Pull request merged**